File: app/src/main/java/com/sanislo/movieapp/di/AppModule.java
package com.sanislo.movieapp.di;

import android.app.Application;
import android.content.Context;

import dagger.Module;
import dagger.Provides;

@Module(includes = {NetworkModule.class, DatabaseModule.class})
public class AppModule {

    @Provides
    Context provideApplicationContext(Application application) {
        return application.getApplicationContext();
    }
}

File: app/src/main/java/com/sanislo/movieapp/di/AppModule.kt
package com.sanislo.movieapp.di

import android.app.Application
import android.content.Context

import dagger.Module
import dagger.Provides

@Module(includes = [NetworkModule::class, DatabaseModule::class])
class AppModule {

    @Provides
    internal fun provideApplicationContext(application: Application): Context {
        return application.applicationContext
    }
}


